src/main/java/com/cleanroommc/groovyscript/core/GroovyScriptTransformer.java

```diff
@@ -2,40 +2,123 @@
             case InvokerHelperVisitor.CLASS_NAME: {
                 ClassWriter classWriter = new ClassWriter(0);
-                new ClassReader(classBytes).accept(new InvokerHelperVisitor(classWriter), 0);
+                new ClassReader(bytes).accept(new InvokerHelperVisitor(classWriter), 0);
                 return classWriter.toByteArray();
             }
             case CachedClassMethodsVisitor.CLASS_NAME: {
                 ClassWriter classWriter = new ClassWriter(0);
-                new ClassReader(classBytes).accept(new CachedClassMethodsVisitor(classWriter), 0);
+                new ClassReader(bytes).accept(new CachedClassMethodsVisitor(classWriter), 0);
                 return classWriter.toByteArray();
             }
             case CachedClassFieldsVisitor.CLASS_NAME: {
                 ClassWriter classWriter = new ClassWriter(0);
-                new ClassReader(classBytes).accept(new CachedClassFieldsVisitor(classWriter), 0);
+                new ClassReader(bytes).accept(new CachedClassFieldsVisitor(classWriter), 0);
                 return classWriter.toByteArray();
             }
             case CachedClassConstructorsVisitor.CLASS_NAME: {
                 ClassWriter classWriter = new ClassWriter(0);
-                new ClassReader(classBytes).accept(new CachedClassConstructorsVisitor(classWriter), 0);
+                new ClassReader(bytes).accept(new CachedClassConstructorsVisitor(classWriter), 0);
                 return classWriter.toByteArray();
             }
             case StaticVerifierVisitor.CLASS_NAME: {
                 ClassWriter classWriter = new ClassWriter(0);
-                new ClassReader(classBytes).accept(new StaticVerifierVisitor(classWriter), 0);
+                new ClassReader(bytes).accept(new StaticVerifierVisitor(classWriter), 0);
                 return classWriter.toByteArray();
             }
         }
-        return classBytes;
+        return transformSideOnly(transformedName, bytes);
+    }
+
+    private byte[] transformSideOnly(String className, byte[] bytes) {
+        SideOnlyConfig.MethodSet bannedProperties = SideOnlyConfig.getRemovedProperties(FMLLaunchHandler.side(), className);
+        if (bannedProperties == null) return bytes;
+        if (bannedProperties.bannsClass) {
+            throw new RuntimeException(String.format("Attempted to load class %s for invalid side %s", className, FMLLaunchHandler.side().name()));
+        }
+
+        ClassNode classNode = new ClassNode();
+        ClassReader classReader = new ClassReader(bytes);
+        classReader.accept(classNode, 0);
+
+        classNode.fields.removeIf(field -> bannedProperties.contains(field.name));
+
+        LambdaGatherer lambdaGatherer = new LambdaGatherer();
+        Iterator<MethodNode> methods = classNode.methods.iterator();
+        while (methods.hasNext()) {
+            MethodNode method = methods.next();
+            if (bannedProperties.contains(method.name + "()")) {
```

Comment:
is there a need to ban by specific method signatures? eg `demo(int)` is client only, but `demo(String)` exists on both.

Reply:
Not sure. Was too lazy to implement